syntax = "proto3";

package envoy.extensions.private_key_providers.sgx.v3alpha;

import "envoy/config/core/v3/config_source.proto";

import "udpa/annotations/status.proto";

option java_package = "io.envoyproxy.envoy.extensions.private_key_providers.sgx.v3";
option java_outer_classname = "SgxProto";
option java_multiple_files = true;
option go_package = "github.com/envoyproxy/go-control-plane/envoy/extensions/private_key_providers/sgx/v3alpha";
option (udpa.annotations.file_status).package_version_status = ACTIVE;

// [#protodoc-title: Sgx private key provider]
// [#extension: envoy.tls.key_providers.sgx]

message SgxPrivateKeyMethodConfig {
  // Path to the ctk library file.
  string sgx_library = 1;

  // ID of the key pair.
  string key_id = 2;

  // User PIN.
  string usr_pin = 3;

  // Security Officer user Pin.
  string so_pin = 4;

  // Label of the token.
  string token_label = 5;

  // Envoy will send CSR object to SDS-Server/Istio-Agent through this channel.
  config.core.v3.ConfigSource sds_config = 6;
}
